gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -103,21 +112,39 @@ private Set<GroupPartOrPackageName> fetchFoldersFromContent(String contents) {
     }
 
     private Set<DependencyVersion> parseVersionsFromContent(String content) {
-        Set<DependencyVersion> versions = new HashSet<>();
         try {
             XmlMapper xmlMapper = new XmlMapper();
+            xmlMapper.registerModule(new GuavaModule());
 
             Metadata metadata = xmlMapper.readValue(content, Metadata.class);
-            if (metadata.versioning() != null && metadata.versioning().versions() != null) {
-                String latest = metadata.versioning().latest();
-                for (String version : metadata.versioning().versions()) {
-                    versions.add(DependencyVersion.of(version, latest.equals(version)));
-                }
+            List<String> allVersions = new ArrayList<>(metadata.versioning().versions());
```

Comment:
this would do well with some tests, it's a bit hard to visualise what you're trying to show here.

To avoid having to serialise and deserialise xml, you make a helper method:

```java
@VisibleForTesting
Set<DependencyVersion> parseVersionsFromContent(Metadata metadata) {
 // ...
}
```

Then you can just make simple `Metadata` objects in your tests and test that things work the way you want

Reply:
it also acts as a way of codifying the ux that you want and prevent regressions are you evolve the ux more etc.

Reply:
Have added some tests so should be covered now